packages/sdk-codegen/src/sdkModels.spec.ts

```diff
@@ -59,1212 +59,1214 @@ import {
-        const expected = [
-          'all_themes',
-          'create_theme',
-          'search_themes',
-          'default_theme',
-          'set_default_theme',
-          'active_themes',
-          'theme_or_default',
-          'validate_theme',
-          'theme',
-          'update_theme',
-          'delete_theme',
-        ]
-        expect(actual).toEqual(expected)
-      })
-    })
-  })
+    // Only change description. Should end up as the same enum
+    const rf4: OAS.SchemaObject = { ...rf, ...{ description: 'RF4' } }
 
-  describe('mayQuote', () => {
-    it('quotes foo-bar', () => {
-      expect(mayQuote('foo-bar')).toEqual(`'foo-bar'`)
-    })
-    it('does not quote foo_bar', () => {
-      expect(mayQuote('foo_bar')).toEqual(`foo_bar`)
-    })
-    it('quotes " foo_bar"', () => {
-      expect(mayQuote(' foo_bar')).toEqual(`' foo_bar'`)
-    })
-    it('does not quote _foo_bar_', () => {
-      expect(mayQuote('_foo_bar_')).toEqual(`_foo_bar_`)
-    })
-  })
+    it('enum types are renamed and not overwritten', () => {
+      const api = new ApiModel({} as OAS.OpenAPIObject)
+      const actual = api.resolveType(rf) as EnumType
+      expect(actual.name).toEqual('ResultFormat')
 
-  describe('camelCase', () => {
-    it.each<[string, string]>([
-      ['', ''],
-      ['foo-bar', 'fooBar'],
-      ['foo -bar', 'fooBar'],
-      ['foo- bar', 'fooBar'],
-      ['foo --  bar', 'fooBar'],
-      ['foo bar', 'fooBar'],
-      ['foo bar--', 'fooBar'],
-      ['foo bar   ', 'fooBar'],
-      ['foo_bar', 'fooBar'],
-      ['foo   bar', 'fooBar'],
-      ['foo -  bar  - - baz', 'fooBarBaz'],
-      ['Foo  -  Bar', 'FooBar'],
-      ['FOOBAR', 'FOOBAR'],
-    ])('"%s" is "%s"', (actual, expected) => {
-      expect(camelCase(actual)).toEqual(expected)
-    })
-  })
+      // Returns first enum for enums with identical values
+      // the `style` parameter is undefined because it is not part of this test. Because TypeScript arguments are
+      // positional only (no named arguments) the parameter must be skipped with an "ignore this" value
+      const actual4 = api.resolveType(rf4, undefined, 'Foo') as EnumType
+      expect(actual4.values).toEqual(actual.values)
+      expect(actual4.name).toEqual('ResultFormat')
+      expect(actual4.description).toEqual(actual.description)
 
-  describe('safeName', () => {
-    it.each<[string, string]>([
-      ['', ''],
-      ['foo-bar', 'foo_bar'],
-      ['foo -bar', 'foo_bar'],
-      ['foo- bar', 'foo_bar'],
-      ['foo --  bar', 'foo_bar'],
-      ['foo bar', 'foo_bar'],
-      ['foo bar--', 'foo_bar_'],
-      ['foo bar   ', 'foo_bar_'],
-      ['foo_bar', 'foo_bar'],
-      ['foo   bar', 'foo_bar'],
-      ['foo -  bar  - - baz', 'foo_bar_baz'],
-      ['Foo  -  Bar', 'Foo_Bar'],
-      ['FOOBAR', 'FOOBAR'],
-    ])('"%s" is "%s"', (actual, expected) => {
-      expect(safeName(actual)).toEqual(expected)
-    })
-  })
+      // Creates unique names
```

Comment:
Without spelunking into the code I would have thought that `name` influenced the generated enum name in models files but it doesn't appear to. How is `name` used?

on second thought, I suppose we just don't have any actual cases in publicly generated code for this? If that's the case, it might be worthwhile adding to the fake section of the test spec and verifying the actual generated output uses a "renamed" name (like `ResultFormat2`)?

Reply:
test spec updated, test added to typescript.gen.spec.ts